lab_2/main.py

```diff
@@ -24,7 +36,15 @@ def get_language_profiles(texts_corpus: list, language_labels: list) -> dict or
     :param language_labels: a list of given language labels
     :return: a dictionary of dictionaries - language profiles
     """
-    pass
+
+    if (not isinstance(texts_corpus, list) or
+            not isinstance(language_labels, list) or not all(isinstance(i, list) for i in texts_corpus) or
+            not all(isinstance(i, str) for i in language_labels)):
+        return None
+    lang_pr = {}
+    for i in range(len(texts_corpus)):
+        lang_pr[language_labels[i]] = get_freq_dict(texts_corpus[i])
```

Comment:
imagine that you have a large text. The creation of a freq dictionary can take time. Do you really want to do that on each iteration?

Reply:
not answered